modules/ardrivo/test/String.cxx

```diff
@@ -0,0 +1,215 @@
+#include <catch2/catch.hpp>
+#include "WString.h"
+#include <string_view>
+
+TEST_CASE("Test compare a string to another" "[compareTo]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "test";
+	String test4 = "est";	
+	String test5 = "Test";
+	REQUIRE(test.compareTo(test2) == 0);
+	REQUIRE(test3.compareTo(test4) == 1);
+	REQUIRE(test5.compareTo(test) == -1);
+}
+
+TEST_CASE("Tests whether or not a String starts with the characters of another String" "[startsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";	
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Tests whether or not a String ends with the characters of another String" "[endsWith]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	String test4 = "test";
+	REQUIRE(test.startsWith(test2));
+	REQUIRE(!test3.startsWith(test4));
+}
+
+TEST_CASE("Copies the String�s characters to the supplied buffer" "[getBytes]")
+{
+	String test = "test";
+	byte arr[5];
+	test.getBytes(arr, test.length());	
+	const char *p = reinterpret_cast<const char*>(arr);
+	REQUIRE_THAT(p, Catch::Contains("test"));
+}
+
+TEST_CASE("Locates a character or String within another String" "[indexOf]")
+{
+	String test = "test";
+	String test2 = "test";
+	String test3 = "est";
+	REQUIRE(test.indexOf(test2) == 0);
+	REQUIRE(test.indexOf(test3) == 1);
+}
+
+TEST_CASE("Modify in place a String removing chars from the provided index to the end of the String or from the provided index to index plus count" "[remove]")
+{
+	String test = "test";
+	test.remove(3);
+	REQUIRE_THAT(test.c_str(), Catch::Contains("tes"));
+}
+
+TEST_CASE("Check if string gets swapped" "[replace]")
+{
+	String test = "ab";
+	String test2 = "cd";
+	test.replace(test, test2);
+	REQUIRE_THAT(test.c_str(), Catch::EndsWith("cd"));
+}
+
+//Crap does not work without std::string
+TEST_CASE("Test if string is given a new capacity" "[reserve]")
+{ 
+	String cap = "";
+	REQUIRE(cap.length() == 0);
+
+	cap.reserve(25);
+	//REQUIRE(cap.length() == 63);
+}
```

Comment:
reserve does not increase the size it only reserves the memory space ([std::basic_string::reserve](https://en.cppreference.com/w/cpp/string/basic_string/reserve)).
You should check if capacity is at least what you reserved -> [https://en.cppreference.com/w/cpp/string/basic_string/capacity](https://en.cppreference.com/w/cpp/string/basic_string/capacity)